hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -402,6 +402,10 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 				return ctrl.Result{}, fmt.Errorf("failed to remove finalizer from hostedcluster: %w", err)
 			}
 		}
+
+		deletionDuration := time.Since(hcluster.DeletionTimestamp.Time).Seconds()
+		clusterDeletionTime.WithLabelValues(hcluster.Name).Set(deletionDuration)
```

Comment:
nit: I would like this logic in a function which receives a hostedCluster e.g `onHostedClusterDeleted(hcluster)` inside `metrics.go` so we can easily change impl/labels in one place in the future.

otherwise, looks great!

Reply:
Thanks! let's don't over abstract for now, we can always do later as we add more things.